libs/core/src/lib/calendar/calendar-header/calendar-header.component.ts

```diff
@@ -212,6 +218,23 @@ export class CalendarHeaderComponent<D> implements OnDestroy, OnInit, OnChanges
     /** Get information about amount of years displayed at once on year view  */
     private _amountOfYearsPerPeriod = 1;
 
+    /**
+     * Function used to disable previous button in the calendar header.
+     */
+
+    @Input()
+    set previousButtonDisableFunction(disableFunc: (date?: D) => boolean) {
+        this.previousButtonDisabled = disableFunc();
```

Comment:
Agree with @platon-rov:
1. Checks should be applied whenever bindings are changed, date is changed or date view is changed
2. As a context it would be better to pass selected date and selected view type 
3. Arguments should be passed to the function
4. You should also check if provided value is actually a function
5. Extract type of the function to a separate type, that can be used outside of the component